ast/expressions.go

```diff
@@ -59,46 +57,13 @@ type ValueExpr struct {
 func NewValueExpr(value interface{}) *ValueExpr {
 	ve := &ValueExpr{}
 	ve.Data = types.RawData(value)
-	// TODO: make it more precise.
-	switch x := value.(type) {
-	case nil:
-		ve.Type = types.NewFieldType(mysql.TypeNull)
-	case bool, int64, int:
-		ve.Type = types.NewFieldType(mysql.TypeLonglong)
-	case uint64:
-		ve.Type = types.NewFieldType(mysql.TypeLonglong)
-		ve.Type.Flag |= mysql.UnsignedFlag
-	case string, UnquoteString:
+	if _, ok := value.(UnquoteString); ok {
```

Comment:
`UnquoteString` is only used for wildcard symbol `*` will not be wrapped in a `DataItem` so it is ok to use value direclty.
And `UnquoteString` is a type in `ast`, package `types` can not handle this type.